client/db/bolt/db.go

```diff
@@ -76,6 +77,9 @@ var _ dexdb.DB = (*BoltDB)(nil)
 
 // NewDB is a constructor for a *BoltDB.
 func NewDB(dbPath string) (dexdb.DB, error) {
+	_, err := os.Stat(dbPath)
+	exists := os.IsNotExist(err)
```

Comment:
missing `!`. 

`exists := !os.IsNotExist(err)`

Or since your negating below, maybe just change the semantics to 

`isNew := os.IsNotExist(err)`